apis/notifications.md

```diff
@@ -129,6 +129,71 @@ Build and run your app again, and press the "Notify" button. Notice that the not
 Remember that `SimpleAction`s added in the `Application` class with `add_action ()` are automatically added in the `app` namespace. Notifications can't trigger actions defined in other namespaces like `win`.
 {% endhint %}
 
+## Default Action
+
+You may have noticed that when you click on a new notification, a new window pops up. This is happening because the notification has a default action that is executed when the user clicks on it. If you don't set it, it will `activate` your application, where we create a new window and present it.
+
+You can change the default action using  [`set_default_action ()`](https://valadoc.org/gio-2.0/GLib.Notification.set\_default\_action.html)
+
+```vala
+notify_button.clicked.connect (() => {
+    var notification = new Notification ("Hello World");
+    notification.set_body ("This is my first notification!");
+    notification.set_default_action ("app.quit");
+
+    send_notification (null, notification);
+});
+```
+
+If you want to avoid creating a new window every time your application is activated, you need to store your main window in an instance variable and check if it has already been shown.
+
+Add an instance variable `main_window` to the top of your class.
+
+{% code title="Application.vala" %}
+```vala
+private Gtk.ApplicationWindow main_window;
+```
+{% endcode %}
+
+Now you can use this variable to store your window.
+
+{% code title="Application.vala" %}
+```vala
+protected override void activate () {
+    if (main_window == null) {
```

Comment:
Gtk.Application already has an `active_window` property so I think we can re-use that instead of storing our own window instance https://valadoc.org/gtk4/Gtk.Application.active_window.html